SpiceSharp/Algebra/Solve/ComplexSolver.cs

```diff
@@ -0,0 +1,333 @@
+﻿using System;
+using System.Numerics;
+using SpiceSharp.Algebra.Solve;
+
+namespace SpiceSharp.Algebra
+{
+    /// <summary>
+    /// Class for solving real matrices
+    /// </summary>
+    public class ComplexSolver : Solver<Complex>
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public ComplexSolver()
+            : base(new Markowitz<Complex>())
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="size">Matrix size</param>
+        public ComplexSolver(int size)
+            : base(new Markowitz<Complex>(), size)
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="size">Size</param>
+        /// <param name="strategy">Strategy</param>
+        public ComplexSolver(int size, PivotStrategy<Complex> strategy)
+            : base(strategy, size)
+        {
+        }
+
+        /// <summary>
+        /// Factor the matrix
+        /// </summary>
+        public override bool Factor()
+        {
+            MatrixElement<Complex> element, column;
+
+            // Get the diagonal
+            element = Matrix.GetDiagonalElement(1);
+            if (element.Value.Equals(0))
+                return false;
+
+            // pivot = 1 / pivot
+            element.Value = Inverse(element.Value);
+
+            // TODO: maybe we should cache this
+            MatrixElement<Complex>[] dest = new MatrixElement<Complex>[Matrix.Size + 1];
+
+            // Start factorization
+            Complex mult;
+            for (int step = 2; step <= Matrix.Size; step++)
+            {
+                // Scatter
+                element = Matrix.GetFirstInColumn(step);
+                while (element != null)
+                {
+                    dest[element.Row] = element;
+                    element = element.Below;
+                }
+
+                // Update column
+                column = Matrix.GetFirstInColumn(step);
+                while (column.Row < step)
+                {
+                    element = Matrix.GetDiagonalElement(column.Row);
+
+                    // Mult = dest[row] / pivot
+                    mult = dest[column.Row].Value * element.Value;
+                    dest[column.Row].Value = mult;
+                    while ((element = element.Below) != null)
+                    {
+                        // dest[element.Row] -= mult * element
+                        dest[element.Row].Value -= mult * element.Value;
+                    }
+                    column = column.Below;
+                }
+
+                // Check for a singular matrix
+                element = Matrix.GetDiagonalElement(step);
+                if (element == null || element.Value.Equals(0.0))
```

Comment:
@svenboulanger element.Value.Equals(0) or element.Value.Equals(0.0) ? I see different above.

Reply:
I believe Equals(0) or Equals(0.0) both resolve to `IEquatable<Complex>.Equals(Complex)`. But'll change it anyway.